syntax = "proto3";

package protocols.decision;

import "protocols/common/geometry.proto";
import "protocols/common/robot_id.proto";

message Decision {
  repeated Behavior behavior = 1;
  TacticalPlan plan = 2;
}

message Behavior {
  enum BehaviorId {
    UNSPECIFIED = 0;
    GOALKEEPER = 1;

    reserved 2;
    reserved "LINE_GOALKEEPER";

    reserved 3;
    reserved "FULL_BACK";

    reserved 4;
    reserved "CENTER_BACK";

    reserved 5;
    reserved "DEFENSIVE_MIDFIELDER";

    reserved 6;
    reserved "OFFENSIVE_MIDFIELDER";

    reserved 7;
    reserved "WINGER";

    FORWARD = 8;
  }

  BehaviorId id = 1;
  common.RobotId robot_id = 2;
  optional common.Point2Df target = 3;
}

message TacticalPlan {
  OffensivePlan offensive = 1;
  DefensivePlan defensive = 2;
}

message DefensivePlan {
  // todo: mapear cálculos que podem ser repassados para o behavior do decision
}

message OffensivePlan {
  // todo: mapear cálculos que podem ser repassados para o behavior do decision
}
